### DISCUSSION TITLE
Problem Merging PDF's

### DISCUSSION THREAD

**Discussion by cruizmib** [gotenberg/gotenberg#860] [Q&A]
I have process that utilizes gotenberg:8.3.0 to convert HTML to PDF and also gather other PDF's, then merge the PDF's to a final PDF. I'm using the REST API endpoint (/forms/pdfengines/merge) to perform the final merge. On the result of the final merge, the PDF is corrupted. As a validity check, I'm using a method to calculate the page count, which inheretly tells me if the PDF is valid. If not valid, this error is thrown:

`Page tree root must be a dictionary`

I'm using this library (org.apache.pdfbox:pdfbox:2.0.24) to validate page count.

When this first begun, I was using version 7.8.1, and initially I thought it was the tagging causing undesirable outcomes. So I upgraded to 8.3.0, and but the issue persist, however not as often. The real head scratcher is that if I were to run the process again, on the same exact files, and same order, it produces a valid final merged PDF. 

Any insight on this would be very helpful. In addition, the other PDF's (that are not being produced by gotenberg) are being produced with Aspose.Words for .NET 24.1.0 as the PDF producer. I'm unsure if that would have any effect on the merging.

**Comment by gulien** [comment]
Hello @cruizmib! 

Try to put the `qpdf` engine in front of other engines, using `—pdfengines-engines=qpdf,libreoffice-pdfengine,exiftool` (https://gotenberg.dev/docs/configuration#pdf-engines).

**Reply by cruizmib** [reply]
Thank you for the prompt reply. Curious, why does this order matter? I presume if not defining a flag, then all is used, the first engine is always used?

**Reply by gulien** [reply]
Exactly 👍 you may see the PDF engines order on startup 😉
